pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -169,9 +169,15 @@ func (c *Cluster) Start() error {
 	// If the mgr is newly created, wait for it to start before continuing with the service and
 	// module configuration
 	for _, d := range deploymentsToWaitFor {
-		if err := waitForDeploymentToStart(c.clusterInfo.Context, c.context, d); err != nil {
+		mgrLabel := fmt.Sprintf("%s=%s", controller.DaemonIDLabel, cephv1.KeyMgr)
+		allRunning, err := waitForDeploymentToStart(c.clusterInfo.Context, c.context.Clientset, c.clusterInfo.Namespace, mgrLabel)
```

Comment:
The new method doesn't wait, it just checks. How about creating a new helper in k8sutil that does wait for them to all be running, and implements the sleep and retry? Anyway, we don't want to sleep in this method or else the unit tests would take too long sleeping.

Reply:
I introduced a new method to wait for all the pods (and UT just returns without waiting).